Kudu.Core/Deployment/Generator/ExternalCommandFactory.cs

```diff
@@ -121,7 +122,14 @@ private string StarterScriptPath
         {
             get
             {
-                return Path.Combine(_environment.ScriptPath, StarterScriptName);
+                if (OSDetecter.IsOnWindows())
+                {
+                    return Path.Combine(_environment.ScriptPath, StarterScriptName);
+                }
+                else
+                {
+                    return "/bin/bash";
```

Comment:
added below comment
````
// Content of StarterScript in windos is "@%*", means to be less verbose when execute a external script,
// since by default cmd will print out content of each step.
// Linux doesn`t print out the content of the step when execute a command, so simply use "/bin/bash" here.
````